sstables/compaction.cc

```diff
@@ -511,7 +519,22 @@ class compaction {
         report_start(formatted_msg);
 
         _compacting = std::move(ssts);
-        return make_sstable_reader();
+
+        auto& cs = _cf.get_compaction_strategy();
+        auto consumer = cs.make_interposer_consumer(_ms_metadata,
```

Comment:
couldn't this introduce a overhead for other strategies that don't need this specialized consumer? @denesb